### DISCUSSION TITLE
Questions about API Routes design ?

### DISCUSSION THREAD

**Discussion by Lizhooh** [vercel/next.js#13931] [Ideas]
Why put the api under the pages directory instead of the root directory?

> Any file inside the folder pages/api is mapped to /api/* and will be treated as an API endpoint instead of a page.

```js 
- myapp
    - pages 
        - api
```

Why not design in the root directory? What is the main reason?

```js 
- myapp
    - api 
    - pages 
```

Hope to get your answer.

**Comment by revskill10** [comment]
Because Vercel support zero-config api routes at root folder, so in case you deploy a Next.js application with api to Vercel, `pages/api` is Next.js specific api routes, not Vercel api routes.

**Comment by jordymeow** [comment]
And is that possible to change? :)
